src/main/java/randoop/generation/GrtFuzzing.java

```diff
@@ -0,0 +1,480 @@
+package randoop.generation;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Executable;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+import randoop.main.GenInputsAbstract;
+import randoop.main.RandoopBug;
+import randoop.operation.CallableOperation;
+import randoop.operation.ConstructorCall;
+import randoop.operation.MethodCall;
+import randoop.operation.TypedClassOperation;
+import randoop.operation.TypedOperation;
+import randoop.sequence.Sequence;
+import randoop.types.NonParameterizedType;
+import randoop.types.Type;
+import randoop.types.TypeTuple;
+import randoop.util.Randomness;
+
+/**
+ * Implements the "GRT Impurity" component, as described in "GRT: Program-Analysis-Guided Random
+ * Testing" by Ma et. al (ASE 2015): https://people.kth.se/~artho/papers/lei-ase2015.pdf.
+ *
+ * <p>The GRT Impurity component (Will be referred to as "GRT Fuzzing") is a fuzzing mechanism that
+ * alters the states of (or creates new) input objects for methods under test to generate more
+ * object states and potentially trigger more branches, improving coverage for the program under
+ * test.
+ *
+ * <p>[TODO: As described in the GRT paper, "GRT Fuzzing" also generates more effective test with
+ * shorter length by reducing the number of redundant sequences that do not side-effect the state of
+ * an object. That is not implemented here.]
+ *
+ * <p>For details about fuzzing, see {@link #fuzz}.
+ */
+@SuppressWarnings("NotJavadoc") // perhaps https://github.com/google/error-prone/issues/3904
+public class GrtFuzzing {
+
+  /**
+   * An enum representing the fuzzing operations for Strings. Each operation has a corresponding
+   * method to be invoked on a StringBuilder object.
+   */
+  private enum StringFuzzingOperation {
+    /** Insert a random character at a random index in the string. */
+    INSERT {
+      @Override
+      Sequence getInputs(int stringLength) {
+        int randomIndex = Randomness.nextRandomInt(stringLength + 1);
+        char randomChar = (char) (Randomness.nextRandomInt(95) + 32); // ASCII 32-126
+        Sequence randomIndexSequence = Sequence.createSequenceForPrimitive(randomIndex);
+        Sequence randomCharSequence = Sequence.createSequenceForPrimitive(randomChar);
+        return Sequence.concatenate(randomIndexSequence, randomCharSequence);
+      }
+
+      @Override
+      List<Executable> getMethods() throws NoSuchMethodException {
+        List<Executable> methodList = new ArrayList<>();
```

Comment:
Throughout this enum, using `Arrays.asList()` is shorter and more efficient than `new ArrayList`.